pkg/config/config.go

```diff
@@ -92,12 +85,20 @@ type Dimension struct {
 	Value string `yaml:"value"`
 }
 
-type Tag struct {
-	Key   string `yaml:"key"`
-	Value string `yaml:"value"`
+type Role struct {
+	RoleArn    string `yaml:"roleArn"`
+	ExternalID string `yaml:"externalId"`
 }
 
-func (c *ScrapeConf) Load(file *string) error {
+func (r *Role) ValidateRole(roleIdx int, parent string) error {
+	if r.RoleArn == "" && r.ExternalID != "" {
+		return fmt.Errorf("Role [%d] in %v: RoleArn should not be empty", roleIdx, parent)
+	}
+
+	return nil
+}
+
+func (c *ScrapeConf) Load(file *string, validSvc func(string) bool) error {
```

Comment:
This refactor stood out, it feels a bit odd to allow a consumer to provide a func which validates the services. I think I would expect a defined way to do that as the default behavior of loading the config. What are your thoughts here?

Reply:
Totally agree. I've resorted to this approach to avoid a dependency loop: almost all other pkgs depend on `config`, including `services`, so couldn't make `config` depend on `services`.

Might move `services.go` into `config` pkg... a bit extreme but might make sense on a level. What do you suggest?

Reply:
Ah yeah I didn't think about the depenency loop 😢 . I think moving `services.go` to the `config` pkg sounds like a good plan for now.

Reply:
Second thought, related to the comment on `aws_tags.go` above. Moving `services.go` to `pkg/config` make the latter depend on `pkg/services` (or whatever other package where `aws_tags.go` is placed). Again a loop.

A possible alternative is a code refactoring where `services.go` is split in something like: all the const stuff (names, aliases, etc) is moved to a struct of some sort in `pkg/config`, while the `{Filter,Resource}Func` and everything else that depends on `aws_tags.go` definitions is kept in `services.go`. (plus some additional changes here and there to deal with the fact `SupportedServices` is being split)

Reply:
👍 makes sense to let this clear and adjust further after